### PR TITLE
Fix: CORS nova_cors_plugin

### PR DESCRIPTION
**Pull request by Holmqvist1990**
# Changes.
* `.gitignore` crashdumps.
* Plugins now in `src/plugins`.
* Capizalited `X-Correlation-ID` in `nova_correlation_plugin`.
* Pre-flight `OPTIONS` succeeds.
* `nova_cors_plugin` now `0.2.0`.
* Custom origins via option `allow_origins`.
* New module `nova_plugin_utilities` parses options.
* Options can now be list values `{K, V} when is_list(V)`:
```
#{
  example_option1 => true, 
  example_option2 => [1, 2, 3]
 } 
```

Note: This assumes `allow_origins` is populated in order to work. I think this makes sense, as defaulting to `*` seems bad practice.

# Fixes.
https://github.com/novaframework/nova/issues/130, if this is desired behaviour.

### CODE REVIEW

**Comment by Taure** [comment]
Stop in the plugin will say that we don't want to continue in the pipeline. So we will not go to next plugin or controller.

What we want here is to check Req for origin and add it to allow-origin. Instead of the *.

With the stop we don't need to implement a options function in each module or have a function clause to handle options.

**Comment by Holmqvist1990** [comment]
@Taure I first wrote it like this:
```
pre_request(#{headers := #{<<"origin">>:= Origin}} = Req, _) ->
    Req1 = cowboy_req:set_resp_header(<<"Access-Control-Allow-Origin">>, Origin, Req),
```

But that provided no change. In that case, the problem lies elsewhere (perhaps a handler).

**Comment by Holmqvist1990** [comment]
@Taure This is with your changes, and it's the only one that works reliably.

For instance, with this version `http_api_demo` works with decoupled frontends :)

**Comment by burbas** [comment]
Just a more general take on this:

This it would be really nice to be able to configure the plugin a bit more dynamically. Something like this perhaps:

```
{pre_request, nova_cors_plugin, #{"allow_origins" => ["my_domain1", "my_domain2"]}
```

Don't know if you'd like to do that in this branch, but it was just an idea I got when reading your commit :)

File: src/plugins/nova_plugin_utilities.erl [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+parse_options(Options) ->
+    ListOptions = maps:to_list(Options),
+    parse_options(ListOptions, []).
+
+parse_options([], Parsed) -> Parsed;
+parse_options([{K, V}|T], Parsed) when V == true ->
+    parse_options(T, [K|Parsed]);
+parse_options([{K, L}|T], Parsed) when is_list(L) ->
+    Strings = lists:map(fun binary_to_list/1, L),
```
Comment (Taure) [inline src/plugins/nova_plugin_utilities.erl:12]: Shouldn't this be other way around? List to binary?

L is a list when entering the function.

**Review by Taure — APPROVED** [review]

**Pull request merged**